### ISSUE TITLE
[bug] Reposts notifications not being sent

### ISSUE THREAD

**Issue by suflors** [superseriousbusiness/gotosocial#3508]
### Describe the bug with a clear and concise description of what the bug is.

When a post is reposted from another instance, it seems that the fact that it has been reposted by someone is not being sent to the server.

### What's your GoToSocial Version?

v0.17.1+git-8a93300

### GoToSocial Arch

amd64 Binary

### What happened?

I made a [post here on my server](https://goto.tomato.pink/@kelsey/statuses/01JBQ6G981SGD2HJ59QP5DTYJ6), and shortly after I was informed by my friend that a big account had [reposted it here](https://app.wafrn.net/fediverse/post/0b06d8d0-9b09-4317-965e-110e4f25f148), however, checking my notifications, I saw nothing. Furthermore, the number of reposts listed on the post has not gone up, either.

### What you expected to happen?

I expected to receive a notification and the repost number to go up.

### How to reproduce it?

Repost a post from a gotosocial instance onto another instance.

### Anything else we need to know?

I directed my friend to repost from a Misskey 2024.10.1 and Hollo 0.1.6 instance as well, and I did not receive notifications for either of these as well (the reposts are now gone as they un-reposted the post.)

**Comment by suflors** [comment]
I've looked at it a bit more and tried some things. Even from an instance that was able to repost something of mine and a notification was sent to me, reposting from the same instance did not notify me, nor did favoriting the post. Furthermore, I see a handful of different warning messages in the logs that look like this:
`Nov 02 20:15:45 Marinette gotosocial[1618]: timestamp="02/11/2024 20:15:45.571" func=streaming.(*Module).readFromWSConn level=WARN streamID=01JBQ89NA7RDAJBBFDBQWNY6EV username=kelsey requestID=8zar9vmj04000xeabza0 msg="unknown 'stream' field: {subscribe user:notification }"`
Not sure if this is the problem happening or not.

**Comment by suflors** [comment]
[This post](https://goto.tomato.pink/@kelsey/statuses/01JBQ8GX0RMXDG5156C72BGJ8D) seems to have gotten 2 favorites, however, my notifications are *still* empty.

**Comment by suflors** [comment]
Suddenly, all of the notifications appeared just now, not exactly sure why.

**Comment by tsmethurst** [comment]
This sounds more like a client problem to me, tbh, given the error you posted. Which client are you using? Re: that wafrn post, isn't it the reply to your post that's been boosted, and not your post itself?

**Comment by suflors** [comment]
Not likely to be the client, as I said it doesn't show on the actual gotosocial itself the interactions (until when I mentioned they randomly appeared). The wafrn post is my own post that's been reposted. Furthermore, I checked on 3 different clients (fedilab, elk, and phanpy), the same was the case on all 3. I'd be incredibly surprised were it not a problem with gotosocial itself thusly.

**Comment by tsmethurst** [comment]
Hmm okay. The only thing I can think of that might be causing this would be a rate limiting issue due to a misconfiguration. In your server logs, do you see the same client-ip value for all incoming connections?

**Comment by tsmethurst** [comment]
Yes, turning off rate limiting would prevent that issue from occurring, though if you're seeing different client IP values then you should leave rate limiting enabled to avoid your instance being overwhelmed.

Beyond that I'm not really sure what might be causing the issue. Could you try updating your instance to 0.17.3 from 0.17.1 just to be sure you're on the latest release version, and see if the problem persists?

**Comment by suflors** [comment]
Updated, will report back.

**Comment by suflors** [comment]
Looked like it was running fine, but woke up today and the last notes were from 10 hours ago. It usually took less time to break (about 6 hours maybe?), the only difference outside of updating is that i limited the cache to 1 day and every 8 hours, and turned back on ratelimiting to the default.

**Comment by suflors** [comment]
If the cache size is something, maybe it's something regarding storage? My instance's server is only a 25gb SSD. Before pruning the cache gotosocial took up 11gb, whereas after the fact only 6gb, it lasts way longer now so I can't imagine it's not significant.

**Comment by tsmethurst** [comment]
Mm, if it's the case that you simply didn't have enough room left on your machine to store a new notification in the database I can imagine that causing problems. But I'd be surprised if that was the issue. Has it persisted?

**Comment by suflors** [comment]
Yes it still happens, though I have set my server to just restart every 4 hours or so to prevent it from happening. The machine has never once been actually full as I've checked it, so I really don't know what it could be.

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/3508.
